feat: created the first pages, navigation on states

TimraWork opened the pull request:
BREAKING CHANGE: Removed the standard structure of create-react-app.

- [ ] Created the simplest components for login, profile, card, registration
- [ ] Added switching of pages by clicking navigation (using the state)
- [ ] Added styles for forms
- [ ] Forms of login and registration
- [ ] Added a handler for submitting the login and registration forms, which redirects by state to the map page

![image](https://user-images.githubusercontent.com/59473768/99632207-a8aaf280-2a5e-11eb-938d-ae0f24b9799a.png)

TimraWork committed:
feat: created the first pages, navigation on states

BREAKING CHANGE: Removed the standard structure of creatre-react-app.

Allarepossible commented on src/components/app.js:
```diff
@@ -0,0 +1,61 @@
+
+import Header from '../components/header';
+
+import PageMap from '../components/page-map';
```
названия файлов лучше называть как и сам компонент, ```../components/PageMap```
Вообще файлы с компонентами, как и сами компоненты называют с большой буквы. так принято)

Allarepossible commented on src/components/app.js:
```diff
@@ -0,0 +1,61 @@
+    navUrl: REDIRECT_URL,
+  };
+
+  handleNavClick = (navUrl, evt) => {
```
тут лучше не менять их местами, и событие можно называть просто ```e```
чтобы всегда событие шло первым аргументом)

Allarepossible commented on src/components/app.js:
```diff
@@ -0,0 +1,61 @@
+
+    return (
+      <div className={ 'container' +
+            ((navUrl === navPath.LOGIN.path || navUrl === navPath.REGISTRATION.path)
```
мне кажется проще будет читать, если вынести в переменную

```const navStyle = (navUrl === navPath.LOGIN.path || navUrl === navPath.REGISTRATION.path) ? ' container--column' : '';```

а если страниц таких будет больше чем два? в таких случаях лучше вынести такие страницы в массив и проверять на наличие текущего урла. Для производительности в таких случаях используют Set, например:
```
const pagesWithoutHeader = new Set([navPath.LOGIN.path, navPath.REGISTRATION.path]);
const navStyle = pagesWithoutHeader.has(navUrl) ? ' container--column' : '';
```

Allarepossible commented on src/components/app.js:
```diff
@@ -0,0 +1,61 @@
+      >
+        <Header handleNavClick={this.handleNavClick} navUrl={navUrl} />
+        <main className="main">
+          {(() => {
```
тут можно упростить немного:
```
{ navUrl === navPath.MAP.path && <PageMap /> }
{ navUrl === navPath.PROFILE.path && <PageProfile handleFormSubmit={this.handleFormSubmit} /> }
...
```

Allarepossible commented on src/components/header.js:
```diff
@@ -0,0 +1,19 @@
+import LogoText from '../assets/img/logo-text.svg';
+
+const HeaderView = ({handleNavClick, navUrl}) => {
+  return (
```
можно не использовать return
```
const HeaderView = ({handleNavClick, navUrl}) => ( 
<header> 
... 
</header>
  );
```

Allarepossible commented on src/components/page-profile.js:
```diff
@@ -0,0 +1,23 @@
+          <div className="form__text">
+            Платёжные данные обновлены. Теперь вы можете заказывать такси.
+          </div>
+          <input
```
вот тут не очень понятно, почему это инпут, и при чем тут форма?

на этой странице по идее 2 состояния может быть:
1. Заполнение данных карты (там используем форму с кнопкой сабмит, которая переводит страницу в состояние 2)
2. Окно с текстом про успешное сохранение, где простая кнопка - ссылка на карту

у тебя получилось немного их смешать)

Allarepossible replied:
но вообще  не было задания реализовывать все компоненты)

ты молодец! крутой задел на следующие домашки!

TimraWork committed:
refactor: add changes with edits from the mentor

rename component file names, simplify code to reading, create Profile component

fix #1

Allarepossible commented:
кстати normilize можно импортить из вне, и не добавлять к себе лишний файл, подробнее тут:
https://create-react-app.dev/docs/adding-css-reset

Allarepossible commented:
все отлично! осталось разрешить все конфликты и мержить ветку в мастер! дз засчитана!

TimraWork committed:
ci: install material, Loft-taxi-mui-theme

import normalize.css from create-react-app

TimraWork committed:
ci: add committizer to package.json

remove normalize, rename card
